File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        unordered_set numsRange(nums.begin(), nums.end());
        vector<int> result;
        for (int i = 1; i <= nums.size(); ++i){
            if (numsRange.find(i) == numsRange.end()){
                result.push_back(i);
            }
        }
        return result;
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for num in nums:
            i = abs(num) - 1
            nums[i] = -1 * abs(nums[i])
        result = []
        for i in range(len(nums)):
            if (nums[i] > 0):
                result.append(i+1)
        return result

        #[4,3,2,7,8,2,3,1]